### PR TITLE
overlord/state: panic in MarkEdge() if task is nil

### PR DESCRIPTION
**Pull request by stolowski**
It was suggested by @pedronis in the other review that `MarkEdge` panics if task that we want to associate with an edge is `nil`. This PR does that.

### CODE REVIEW

File: overlord/state/task.go [hunk 1]
```diff
@@ -522,6 +522,9 @@ func (ts *TaskSet) AddTask(task *Task) {
 // MarkEdge marks the given task as a specific edge. Any pre-existing
 // edge mark will be overridden.
 func (ts *TaskSet) MarkEdge(task *Task, edge TaskSetEdge) {
+	if task == nil {
```
Comment (bboozzoo) [inline overlord/state/task.go]: A unit test would be nice.

Comment (stolowski) [reply overlord/state/task.go]: Added. It actually uncovered a mistake in panic string 😯

**Comment by stolowski** [comment]
> Looks fine! A (unit) test for this might be nice but its not super important (it's pretty obvious)

Huh, it actually uncovered an issue...

File: overlord/state/task.go (hunk 1 above)
Comment (bboozzoo) [reply overlord/state/task.go]: Hah, right `%q` missing format.

**Pull request merged**